source/dofs/dof_tools_sparsity.cc

```diff
@@ -983,21 +983,37 @@ namespace DoFTools
         std::vector<types::global_dof_index> dofs_on_this_cell(DoFTools::max_dofs_per_cell(dof));
         std::vector<types::global_dof_index> dofs_on_other_cell(DoFTools::max_dofs_per_cell(dof));
 
-        std::vector<Table<2,Coupling> > int_dof_mask (fe.size());
-
-        int_dof_mask = dof_couplings_from_component_couplings (fe, int_mask);
-
-        // Convert the int_dof_mask to bool_int_dof_mask so we can pass it
-        // to constraints.add_entries_local_to_global()
-        std::vector<Table<2,bool> > bool_int_dof_mask (fe.size());
+        const unsigned int n_components = fe.n_components();
+        AssertDimension (int_mask.size(0), n_components);
+        AssertDimension (int_mask.size(1), n_components);
+        AssertDimension (flux_mask.size(0), n_components);
+        AssertDimension (flux_mask.size(1), n_components);
+
+        Table<2,Coupling> int_and_flux_mask (n_components, n_components);
+        for (unsigned int c1=0; c1<n_components; ++c1)
+          for (unsigned int c2=0; c2<n_components; ++c2)
+            if (int_mask(c1,c2) != none || flux_mask(c1,c2) != none)
+              int_and_flux_mask(c1,c2) = always;
+
+        std::vector<Table<2,Coupling> > int_and_flux_dof_mask (fe.size());
+
+        int_and_flux_dof_mask = dof_couplings_from_component_couplings (fe, int_and_flux_mask);
```

Comment:
Just remove the initializer in line 998 and use the copy or move constructor here. Also, if you put a comment in front of line 992, then we'd remember why you do it this way.